### PR TITLE
[ROCm] Add ROCm support for batch_kernels

### PR DESCRIPTION
**Pull request by jerryyin**

### CODE REVIEW

**Review by whchung — DISMISSED** [review]
rebase please

File: tensorflow/core/kernels/concat_lib.h [hunk 1]
```diff
@@ -47,33 +46,15 @@ void ConcatCPU(
...
-TF_CALL_complex128(REGISTER);
-TF_CALL_int32(REGISTER);  // Needed for TensorLists.
-TF_CALL_int64(REGISTER);
-TF_CALL_int16(REGISTER);
-TF_CALL_bfloat16(REGISTER);
-TF_CALL_bool(REGISTER);
-TF_CALL_uint8(REGISTER);
-#undef REGISTER
-#endif  // GOOGLE_CUDA
+#endif  // GOOGLE_CUDA || TENOSORFLOW_USE_ROCM
```
[the remark this answers is not in the document]
Comment (jerryyin) [reply tensorflow/core/kernels/concat_lib.h]: Sounds good

**Comment by jerryyin** [comment]
I signed it!

**Comment by meteorcloudy** [comment]
Sorry I don't think I'm the right person to review all the ROCM changes, I don't have enough knowledge about how it works.

**Review by gunan — APPROVED** [review]

**Pull request merged**